### ISSUE TITLE
What is the meaning of MomentumComponent derived attribute?

### ISSUE THREAD

**Issue by sbastrakov** [ComputationalRadiationPhysics/picongpu#4142]
This is a derived field available in the default `fileOutput.param`, so in a way it's available (but not enabled) by default. However, its meaning and intended use case feels not clear to me. There is [this description](https://github.com/ComputationalRadiationPhysics/picongpu/blob/74b6e552f7a436d517574b93c35c17830619994d/include/picongpu/particles/particleToGrid/derivedAttributes/MomentumComponent.def#L41-L50) that matches the [operations performed for each macroparticle](https://github.com/ComputationalRadiationPhysics/picongpu/blob/74b6e552f7a436d517574b93c35c17830619994d/include/picongpu/particles/particleToGrid/derivedAttributes/MomentumComponent.hpp#L43-L58).

However, then when calculating a derived field each value will be the sum of those ratios for all contributing macroparticles also taking account their shape, and what is the meaning of the sum? For example, imagine we are calculating it for x component (so template parameter 0) and there are two macroparticles at same position with different weights, one going strictly in +x and the other strictly in -x. By how it's implemented, first one will give +1 and second -1, so that the total value will be 0. However, if the weights are different should it actually be zero, as the average momentum in x is not 0? So roughly speaking we are doing a term-wise ratio and then sum, while I would have expected we first sum and then do a ratio of the two sums.

Basically, my question comes from trying to apply it to calculate an average component of momentum (as was asked by a user), then trying to use `MomentumComponent` and realizing it doesn't fit, and then wondering what does it do at all. It could well be i'm missing the context of its application, then it may point out to lacking documentation there. If anyone has experience using this attribute, please share.

P.S. this attribute also doesn't correctly interact with averagizing, but i think it's a problem in the averagizing, i will make a separate issue #4143 for that.

cc @steindev

**Comment by pordyna** [comment]
Oh, this is weird. I would also expect  calculating the total momentum in 3D and the total momentum in a certain direction and then take the ratio. But this is actually sth. that can be done in a single attribute only with a combined attribute, which is a new feature.

**Comment by sbastrakov** [comment]
@pordyna do you mean to make a new "simple" attribute for a component of momentum that for each macroparticle will return `weight * px` and then use `AverageAttribute` of that to get average value of `px` as a field?

**Comment by sbastrakov** [comment]
So upon looking at the PR that added (the predecessor of) this output https://github.com/ComputationalRadiationPhysics/picongpu/pull/1481 I am still confused about the current implementation. My earlier example with different weights could be even further simplified to this: weights are same, macroparticle momentums are `p1 = (3, 0, 4)`, `p2 = (-1, 0, 0)`. Then with the current algorithm for `MomentumComponent<0>` the contribution of the first macroparticle is `3/sqrt(25) = 0.6` and of the second `-1` for a total of `-0.4`. While the macroparticles combined have a total momentum of `(2, 0, 4)` and so I would have expected the total momentum component to be `2/sqrt(20) = 0.44`.

I think this may be just me misinterpreting the purpose of this output as I've never used it.

**Comment by sbastrakov** [comment]
cc @ax3l if you have time for it

**Comment by sbastrakov** [comment]
`MomentumComponent<>` is now removed, there is a new attribute `Momentum<>` to get agregated values of separate momentum components.
